File: scraping.py
import json
import requests
from bs4 import BeautifulSoup

url = 'https://atcoder.jp/users/Itsuki_54/history'
response = requests.get(url)
html_content = response.content

soup = BeautifulSoup(html_content, 'html.parser')

rows = soup.find_all('tr', class_='text-center')

data_list = []

for row in rows:
    date = row.find('td', {'data-order': True}).text.strip()
    contest_name = row.find('a').text.strip()
    rank = row.find_all('td')[2].text.strip()
    submissions = row.find_all('td')[3].text.strip()
    rating = row.find_all('td')[4].text.strip()

    data_list.append({
        'date': date,
        'contest_name': contest_name,
        'rank': rank,
        'submissions': submissions,
        'rating': rating
    })

    with open('src/data/contest.json', 'w') as f:
        json.dump(data_list, f, indent=4, ensure_ascii=False)

File: src/lib/scraping.js
import axios from "axios";
import * as cheerio from "cheerio";
import { writeFileSync } from "fs";

const url = "https://atcoder.jp/users/Itsuki_54/history";

axios.get(url)
  .then(response => {
    const htmlContent = response.data;

    const $ = cheerio.load(htmlContent);
    const rows = $("tr.text-center");

    const dataList = [];

    rows.each((index, row) => {
      const date = $(row).find("td[data-order]").text().trim();
      const contestName = $(row).find("a").text().trim();
      const rank = $(row).find("td").eq(2).text().trim();
      const submissions = $(row).find("td").eq(3).text().trim();
      const rating = $(row).find("td").eq(4).text().trim();

      dataList.push({
        date: date,
        contest_name: contestName,
        rank: rank,
        submissions: submissions,
        rating: rating,
      });
    });

    const objectContent = `export const contests = ${JSON.stringify(dataList, null, 4)};`;

    writeFileSync("src/data/contest.ts", objectContent, "utf8");
    console.log("Data successfully scraped and saved to src/data/contest.js");
  })
  .catch(error => {
    console.error("Error fetching the URL:", error);
  });
